File: GeeksForGeeks/Pair with given sum in a sorted array/answer.java
class Solution{
    
  
    int Countpair(int a[], int n, int sum)
    {
        int i = 0, j = n-1, count = 0;

        while(i<j){

            // If the two element's sum is equal to sum return its index
            if(a[i] + a[j] == sum){
                count++; i++; j--;
            }
            // If the two element's sum is greater than sum 
            // decrease the second pointer
            else if(a[i] + a[j] > sum){
                j--;
            }
            // If the two element's sum is less than sum 
            // increase the first pointer
            else{
                i++;
            }
        }
        return (count == 0)? -1:count;
    }
  
    
}

File: Leet Code/Split a String in Balanced Strings/answer.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int count = 0, R = 0, L = 0;
        for(auto x:s){
            if(x == 'R'){
                R++;
            }
            else if(x == 'L'){
                L++;
            }

            if(R == L){
                R = L = 0;
                count++;
            }
        }
        return count;
    }
};
